### PR TITLE
chore(react-dialog): updates stories

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. Updates stories organisation
2. Adds all compound components properties tables in the documentation
3. Merges `DialogTitle` stories with `Dialog` stories
4. Adds comments on `no focusable element` scenarios about Talkback issue https://github.com/microsoft/fluentui/issues/24471
5. Adds comments to `DialogOpenChangeEventHandler` inside `DialogProps`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/2b942c5e8bfbb79f8c2af8e97206f7c62380b750'>2b942c5e8bfbb79f8c2af8e97206f7c62380b750</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/Avatar-mount_base.html">1324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/Avatar-mount.html">1313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/Button-mount_base.html">951</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/Button-mount.html">952</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProvider-mount_base.html">1571</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProvider-mount.html">1602</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-mount_base.html">643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-mount.html">625</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">599</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">635</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/MakeStyles-mount_base.html">1878</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/MakeStyles-mount.html">1894</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/SpinButton-mount_base.html">2510</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25070/perf-test-react-components/SpinButton-mount.html">2543</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-dialog-bbb01d69-5227-40c3-aaf3-053f1af35f0d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "chore: updates dialog stories",
```
Comment (layershifter) [inline change/@fluentui-react-dialog-bbb01d69-5227-40c3-aaf3-053f1af35f0d.json:3]:
```suggestion
  "comment": "chore: cleanups in types",
```

_nit:_ We add into changelogs only changes that affects published code.

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-dialog/src/stories/Dialog/index.stories.tsx [hunk 2]
```diff
@@ -14,15 +15,23 @@ export { ChangeFocus } from './DialogChangeFocus.stories';
...
 export default {
-  title: 'Preview Components/Dialog/Dialog',
+  title: 'Preview Components/Dialog',
   component: Dialog,
+  subcomponents: {
+    DialogTrigger,
+    DialogSurface,
+    DialogTitle,
+    DialogActions,
+  },
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/stories/Dialog/index.stories.tsx:29]: 😮 ❤️

Comment (bsunderhus) [reply packages/react-components/react-dialog/src/stories/Dialog/index.stories.tsx:29]: Yeah, I was amazed when I found out about this! 🤩

**Pull request merged**